app/views/propertyrepresentation/appoint/checkYourAnswers.scala.html

```diff
@@ -20,14 +20,26 @@
 
 @import config.ApplicationConfig
 @import models.propertyrepresentation.AppointNewAgentSession
+@import models.propertyrepresentation.AppointmentScope.ALL_PROPERTIES
+@import models.propertyrepresentation.AppointmentScope.RELATIONSHIP
+@import models.propertyrepresentation.All
+@import models.propertyrepresentation.AppointAgentRequest
 
-@this(govukTable : GovukTable, mainLayout: views.html.mainLayout)
+@this(govukErrorSummary : GovukErrorSummary, govukButton : GovukButton, govukTable : GovukTable, mainLayout: views.html.mainLayout)
 
-@(data: AppointNewAgentSession)(implicit request: Request[_], messages: Messages, config: ApplicationConfig)
+@(form: Form[AppointAgentRequest], data: AppointNewAgentSession)(implicit request: Request[_], messages: Messages, config: ApplicationConfig)
 
 @answerMessageKey = @{
     s"propertyRepresentation.checkYourAnswers.options.${data.managingProperty.getOrElse("")}"
 }
+@appointmentScope = @{
+    data.managingProperty match {
+        case None => throw new IllegalStateException("Unable to get 'managingProperty' from session data")
```

Comment:
Isn't this the kind of thing we can avoid by using a request wrapper that guarantees this data is available?

Reply:
i think we might want to blow up if the data is not there at this point in the journey as it would mean that something is wrong. May be @DanielConnelly wants to weigh in on this

Reply:
It would be better if we never got this far if we didn't have the required data. Feels like we need a new ActionFilter (or similar) that ensures that agent data is available? But yes, get @DanielConnelly 's opinion.

Reply:
We should never knowingly allow a 500 exception. it is easier to manage these errors in what darrell suggested. a filter with a new request type would allow you to be sure only those get in correctly.